skunkbooth/main.py

```diff
@@ -43,6 +44,18 @@ def global_shortcuts(event: Event) -> None:
             raise StopApplication("User pressed quit")
 
 
+def saving_video(screen: Screen, vidBuf: Queue) -> None:
+    """Event handler to initiate video saving"""
+    effects = [
+        Cycle(screen, FigletText("SAVING", font="big"), screen.height // 2 - 8),
+        Cycle(screen, FigletText("VIDEO!", font="big"), screen.height // 2 + 3),
+        Stars(screen, (screen.width + screen.height) // 2),
+    ]
+    screen.play([Scene(effects, 100)], repeat=False)
```

Comment:
This currently plays the animation for a fixed length of time regardless of the load, and then it looks like it checks a single time if there's still video left to save. This means that the animation will play for too long even if there's only milliseconds of remaining work to be done, and if there's minutes of work left the animation should finish before we're actually done saving the video. The animation needs check that the workload is done, then terminate.